minutes/2018-10-12-node-interactive.md

```diff
@@ -0,0 +1,199 @@
+
+* maintains create-react-app.
+* interested in getting involved in tooling working group.
+
+### Guy Bedford (@guybedord)
+
+* contributes to modules working group.
+* maintains rollup, systemjs.
+
+### Oktavianus Ludirio (@oludiro)
+
+* works at Zillow.
+* interested in learning more about writing tooling in Node.js.
+
+### Daniel Stockman (@evocateur)
+
+* works at Zillow.
+* maintains Lerna.
+
+### Erick Wendel (@erickwendel)
+
+* Microsoft MVP.
+* curious about tooling (would like to contribute more).
+
+### Ruy Adorno (@ruyadorno):
+
+* contributor to yeoman.
+
+### Michel Lopez (@michellopez):
+
+* senior program manager at Microsoft.
+* interested in making sure Node.js/npm are stable on Windows.
+* excited that Windows and Linux build times are now equivilent.
+
+### Bryan English (@bengl)
+
+* engineer at Intrinsic.
+* interested in intersection between tooling and security (interesting challenges).
+
+## Introduction/Mission
+
+* *Chris*: if you had to write a bundler, or scaffolding tool, how could APIs
+  with Node.js make your life easier for you.
+  * ran user feedback session that indicated there was room for improvement
+    in some APIs.
+
+## Initial Successes
+
+* `mkdirp`:
+  * just landed in Node.js core.
+  * process of landing was a bit bumpy, due to questions around feature
+    detection.
+* `readdir`:
+  * you can now fetch stat information without crossing the JavaScript C++
+    boundary multiple times.
+* V8 Coverage:
+  * you can now dump coverage information directly from V8.
+  
+### What We Learned
+
+* *Ben*: we've had some good successes, but we learned there were some
+  shortcomings around feature detection -- maybe we need to formalize how we
+  propose features to work on ... brings us to.
+  
+## Backwards Compatibility and Polyfills
+
+* *Chris*:
+  * want feature detection in Node.js, but don't want to litigate it right now.
+  * for time being, perhaps we can get away with `process.version`.
+  * good start would be making pulls to community:
+    * update `mkdirp`, `rimraf`, based on `process.version`.
+    * _Michel_ also advocates this approach.
+
+* *Daniel*:
+  * how do we get adoption if only cutting edge Node.js should be using?
+  * perhaps backporting should be goal when possible?
+
+* *Bryan/Chris*:
+  * backporting is a pain in the butt, perhaps we just agree to polyfill;
+    see: readable stream.
+
+* *Ruy*:
+  * there are good aspects to not backporting features, drive more people
+    to adopt new Node.js versions.
+
+* *Ian*:
+  * because we're targetting CLI tools, perhaps we can be more aggressive
+    about pushing people to upgrade to new Node.js:
+    
+* *Daniel*:
+  * on topic of polyfills, `graceful-fs` is an example of 
+    _how not to polyfill_.
+
+* *Guy*:
+  * fs-extra is a better approach, doesn't mutate global fs object
+
+we seem to have reached relative consensus that:
+
+* we'll shim old library versions.
+* won't backport to old Node.js versions.
```

Comment:
LTS generally decides this, don't they? And they do backport semver-minor features, at least if they don't look risky . /cc @nodejs/lts

Reply:
Dropping support of old nodes doesn’t push people to upgrade node - it prevents them from using your tool.

If they’re not on the latest node, it’s because they *cant be*. Not because they haven’t bothered yet.

Reply:
@ljharb the context that many tools are used on developer machines where devs aren’t restricted to certain versions, especially in the cases where Node is only used for, say, bundling g.

Reply:
A machine being a dev machine doesn’t mean that; often, engs at a company are forced to use the same version of node as production, which may be far olde than “latest”.

Reply:
@ljharb that assumes Node is in production and not just a dev tool which poops or a web site. anyway, this doesn’t have much bearing on the initiatives rn.

Reply:
let's create issues for each of the points raised by @sam-github, @ljharb, and @guybedford, all seem like areas where we need more discussion.

Reply:
I don't recall a consensus here, but I do remember simply pointing out that backports can sometimes be difficult, especially when there are huge refactors in between.

Realistically, [it *is* up the the Release group](https://github.com/nodejs/Release#release-plan):

> Once a major version enters LTS coverage, new features (semver-minor) may only be landed with consent of the Release working group.

Anyways, I think we should land these minutes basically as is, and come to a more well-defined consensus later. Minutes are non-binding.